### PR TITLE
adding support for using env. variables in inputs JSON

### LINKED ISSUE #705: Unable to use ENV VARS in input JSON
I tried the following without success:
```
inputs:
   type: json
   MyKey: {
     username: ${username},
     password: ${password}
   }
```

### PR DESCRIPTION
**Pull request by pritidesai**
Closes #705

### CODE REVIEW

File: parsers/manifest_parser_test.go [hunk 1]
```diff
@@ -304,7 +304,7 @@ func TestParseManifestForMultiLineParams(t *testing.T) {
 		assert.Equal(t, expectedResult, actualResult, TEST_MSG_ACTION_FUNCTION_RUNTIME_MISMATCH)
 
 		// test # input params
-		expectedResult = strconv.FormatInt(10, 10)
+		expectedResult = strconv.FormatInt(13, 10)
```
Comment (mrutkows) [inline parsers/manifest_parser_test.go]: 3 new unit tests.

File: parsers/manifest_parser_test.go [hunk 2]
```diff
@@ -1032,7 +1054,7 @@ func TestParseManifestForJSONParams(t *testing.T) {
 		assert.Equal(t, expectedResult, actualResult, TEST_MSG_ACTION_FUNCTION_RUNTIME_MISMATCH)
 
 		// validate the number of inputs to this action
-		expectedResult = strconv.FormatInt(6, 10)
+		expectedResult = strconv.FormatInt(8, 10)
```
Comment (mrutkows) [inline parsers/manifest_parser_test.go]: 2 new unit tests

File: parsers/parameters.go [hunk 3]
```diff
@@ -322,7 +325,7 @@ func ResolveParameter(paramName string, param *Parameter, filePath string) (inte
 	}
 
 	// JSON - Handle both cases, where value 1) is a string containing JSON, 2) is a map of JSON
-	if param.Type == "json" {
+	if param.Value != nil && param.Type == "json" {
```
Comment (mrutkows) [inline parsers/parameters.go]: good catch.

File: tests/dat/manifest_validate_multiline_params.yaml [hunk 4]
```diff
@@ -32,6 +32,15 @@ packages:
...
                         value: foo
+                    param_json_type_and_value_only_1:
+                        type: json
+                        value: '{ "name": "Sam", "place": "Shire" }'
+                    param_json_type_and_value_only_2:
+                        type: json
+                        value: { "name": "${USERNAME}", "password": "${PASSWORD}" }
+                    param_json_type_and_value_only_3:
+                        type: json
+                        value: { "name": "$${USERNAME}", "password": "$${PASSWORD}" }
```
Comment (mrutkows) [inline tests/dat/manifest_validate_multiline_params.yaml]: Thanks for this.  Did we open an spec/docs issue to document this new syntax for string literals (ala Kube, etc.)?

File: wskenv/environment.go [hunk 5]
```diff
@@ -63,15 +63,21 @@ func InterpolateStringWithEnvVar(key interface{}) interface{} {
 				return c == '$' || c == '{' || c == '}'
 			}
 			for _, substr := range strings.FieldsFunc(keystr, f) {
-				//if the substr is a $ENV_VAR
-				if strings.Contains(keystr, "$"+substr) {
+				//if the substr is a $${ENV_VAR}
+				// return ${ENV_VAR}
+				if strings.Contains(keystr, "$${"+substr+"}") {
+					keystr = strings.Replace(keystr, "$${"+substr+"}", "${"+substr+"}", -1)
+					//if the substr is a $ENV_VAR
```
Comment (mrutkows) [inline wskenv/environment.go]: hate gofmt and comment alignment

File: tests/dat/manifest_validate_multiline_params.yaml (hunk 4 above)
Comment (pritidesai) [reply tests/dat/manifest_validate_multiline_params.yaml]: yup we have issue #769 and #770 to update spec and doc respectively.

**Pull request merged**